backend/src/realtime/realtime-note/realtime-note.ts

```diff
@@ -68,8 +70,13 @@ export class RealtimeNote extends EventEmitter2<RealtimeNoteEventMap> {
       } clients left.`,
     );
     this.emit('clientRemoved', client);
-    if (!this.hasConnections() && !this.isClosing) {
-      this.destroy();
+
+    if (this.canBeDestroyed()) {
+      setTimeout(() => {
+        if (this.canBeDestroyed()) {
+          this.destroy();
+        }
+      }, LIFETIME_WITHOUT_CLIENTS);
```

Comment:
I'm not completely sure, but this looks like it could break due to the timeout not being cancelled on reconnects.

Let's look at the following scenario: A note is loaded with one client connected. The client disconnects and the timer starts to run. Within less than 3 seconds, the client re-connects and disconnects again. Now there are two timers running. As there is no client left, the realtime-note gets destroyed by the first timer. Shortly after, the second timer calls this.destroy() on the already destroyed note.

Reply:
Thanks for your review! !

In the case of the scenario you presented, it certainly doesn't work as expected.
Thank you for making me realize that 😄

Reply:
@ErikMichelson
Added a listener that cancels the timer when a client connects.
I think it would work in your scenario as well.